grails-app/domain/com/unifina/domain/security/IntegrationKey.groovy

```diff
@@ -17,6 +22,13 @@ class IntegrationKey implements Serializable {
 	Date dateCreated
 	Date lastUpdated
 
+	@PostConstruct
+	void init() {
+		String password = grailsApplication.config["streamr"]["encryption"]["password"]
+		Assert.notNull(password, "streamr.encryption.password not set!")
+		encryptor = new StringEncryptor(password)
+	}
+
```

Comment:
@hpihkala What is the correct way to get the decryption password in IntegrationKey.groovy? I tried this idiom, which works in EthereumIntegrationKeyService.groovy to inject the grailsApplication bean, but it doesnt work here, maybe bc IntegrationKey.groovy isn't a service.

Reply:
should StringEncryptor be a transient member var or created in jsonMap()?

Reply:
EthereumIntegrationKeyService keyService = Holders.getApplicationContext().getBean(EthereumIntegrationKeyService.class)
keyService.decryptPrivateKey(getValue());
?